crates/adapterlib/src/transport.rs

```diff
@@ -53,7 +53,10 @@ pub trait TransportInputEndpoint: InputEndpoint {
 }
 
 pub trait IntegratedInputEndpoint: InputEndpoint {
-    fn open(&self, input_handle: &InputCollectionHandle) -> AnyResult<Box<dyn InputReader>>;
+    fn open(
+        self: Box<Self>,
```

Comment:
is it common to use `self` for something that does not have `Self` type?

Reply:
There's a small set of types that can be used for `self`, `Box<Self>` is one of them.